nautobot_ssot/migrations/0005_ssotservicenowconfig.py

```diff
@@ -0,0 +1,72 @@
+# Generated by Django 3.2.16 on 2023-06-13 09:15
+
+from django.contrib.contenttypes.models import ContentType
+from django.db import migrations, models
+from django.db.migrations.recorder import MigrationRecorder
+
+import django.db.models.deletion
+import uuid
+
+
+_APP_LABEL = "nautobot_ssot"
+_OLD_APP_LABEL = "nautobot_plugin_ssot_servicenow"
+_MODEL_NAME = "SSOTServiceNowConfig"
+
+
+def _move_data(apps, schema_editor):
+    old_migration = {
+        "app": _OLD_APP_LABEL,
+        "name": "0001_initial",
+    }
+    if not MigrationRecorder.Migration.objects.filter(**old_migration).exists():
+        return
+
+    new_model_name = _MODEL_NAME.lower()
+    new_model = apps.get_model(_APP_LABEL, new_model_name)
+    new_table_name = new_model._meta.db_table
+    old_table_name = new_table_name.replace(f"{_APP_LABEL}_", f"{_OLD_APP_LABEL}_")
+
+    with schema_editor.connection.cursor() as cursor:
+        # Table names are from trusted source (this script)
+        cursor.execute(f"INSERT INTO {new_table_name} SELECT * FROM {old_table_name};")  # nosec
+
+    # Update the content type to point to the new model
+    old_content_type = ContentType.objects.get(app_label=_OLD_APP_LABEL, model=_MODEL_NAME.lower())
+    old_content_type.app_label = _APP_LABEL
+    old_content_type.model = new_model_name
+    old_content_type.save()
+
+    with schema_editor.connection.cursor() as cursor:
+        cursor.execute(f"DROP TABLE {old_table_name} CASCADE;")
```

Comment:
Needs manual testing, should be enough IMHO, as this code is only for backward compatibility purposes. This code runs only when migrating data from old nautobot-plugin-ssot-servicenow.

Reply:
One upside of automated testing would be that once this migration breaks we can identify that in the documentation and say something like

> if you need to migrate from the standalone plugin and your Nautobot version is >=3.0 (example) you need to go through version 1.5 (example)

Maybe with a django upgrade in the future to a new major version this could be necessary?

Reply:
This is more integration than unit testing, because to test it, it's necessary to install, and configure old plugin. Do we want to put the old plugin to dev dependencies here? Does that worth?

Reply:
Maybe we could compromise on a manual data dump from the old plugin?

Reply:
Would be possible. Again, does that worth?

Reply:
Actually such case should not happen IMHO as the old plugin is dependent on nautobot 1.x:

https://github.com/nautobot/nautobot-plugin-ssot-servicenow/blob/develop/pyproject.toml#L23